addons/arsenal/functions/fnc_handleStats.sqf

```diff
@@ -0,0 +1,333 @@
+/*
+ * Author: Alganthe
+ * Handles the stats control group
+ *
+ * Arguments:
+ * 0: Arsenal display <DISPLAY>
+ * 1: Current panel control <CONTROL>
+ * 2: Current panel selection <SCALAR>
+ * 3: Item config entry <CONFIG>
+ *
+ * Return Value:
+ * None
+ *
+ * Public: No
+*/
+#include "script_component.hpp"
+#include "..\defines.hpp"
+params ["_display", "_control", "_curSel", "_itemCfg"];
+
+private _statsBoxCtrl = _display displayCtrl IDC_statsBox;
+private _statsPreviousPageCtrl = _display displayCtrl IDC_statsPreviousPage;
+private _statsNextPageCtrl = _display displayCtrl IDC_statsNextPage;
+private _statsCurrentPageCtrl = _display displayCtrl IDC_statsCurrentPage;
+
+private _hideUnusedFnc = {
+    params ["_numbers"];
+
+    {
+        private _statsTitleCtrl = _display displayCtrl (5101 + ((_x - 1) * 4));
+        private _statsTitleIDC = ctrlIDC _statsTitleCtrl;
+
+        private _statsBackgroundCtrl = _display displayCtrl (_statsTitleIDC + 1);
+        private _statsBarCtrl = _display displayCtrl (_statsTitleIDC + 2);
+        private _statsTextCtrl = _display displayCtrl (_statsTitleIDC + 3);
+
+        {
+            _x ctrlSetFade 1;
+            _x ctrlCommit 0;
+        } forEach [
+            _statsTitleCtrl,
+            _statsBackgroundCtrl,
+            _statsBarCtrl,
+            _statsTextCtrl
+        ];
+    } forEach _numbers;
+};
+
+if !(isNil "_itemCfg") then {
+
+    private _handleStatsFnc = {
+        params ["_statsIndex", "_leftPanel"];
+        private ["_statsArray", "_statsList", "_isLeftPanel", "_currentPage"];
+
+        // Get the proper list and page
+        if (_leftPanel) then {
+            _isLeftPanel = true;
```

Comment:
you could return an array from the if statement here. And then use `params` on that array. Get's rid of that `private`

Reply:
totally true, plus I already do it somewhere else, I think it's for the loadouts list caching